metrics.go

```diff
@@ -33,63 +33,61 @@ type Metrics struct {
 	TracesJoinedNotSampled metrics.Counter `metric:"traces" tags:"state=joined,sampled=n"`
 
 	// Number of sampled spans started by this tracer
-	SpansStarted metrics.Counter `metric:"spans" tags:"group=lifecycle,state=started"`
+	SpansStartedSampled metrics.Counter `metric:"spans_started" tags:"sampled=y"`
 
-	// Number of sampled spans finished by this tracer
-	SpansFinished metrics.Counter `metric:"spans" tags:"group=lifecycle,state=finished"`
+	// Number of unsampled spans started by this tracer
+	SpansStartedNotSampled metrics.Counter `metric:"spans_started" tags:"sampled=n"`
 
-	// Number of sampled spans started by this tracer
-	SpansSampled metrics.Counter `metric:"spans" tags:"group=sampling,sampled=y"`
-
-	// Number of not-sampled spans started by this tracer
-	SpansNotSampled metrics.Counter `metric:"spans" tags:"group=sampling,sampled=n"`
+	// Number of spans finished by this tracer
+	SpansFinished metrics.Counter `metric:"spans_finished"`
 
 	// Number of errors decoding tracing context
-	DecodingErrors metrics.Counter `metric:"decoding-errors"`
+	DecodingErrors metrics.Counter `metric:"span_context_decoding_errors"`
 
 	// Number of spans successfully reported
-	ReporterSuccess metrics.Counter `metric:"reporter-spans" tags:"state=success"`
+	ReporterSuccess metrics.Counter `metric:"reporter_spans_reported"`
 
-	// Number of spans in failed attempts to report
-	ReporterFailure metrics.Counter `metric:"reporter-spans" tags:"state=failure"`
+	// Number of spans not reported due to some failure
+	ReporterFailure metrics.Counter `metric:"reporter_spans_failed"`
 
 	// Number of spans dropped due to internal queue overflow
-	ReporterDropped metrics.Counter `metric:"reporter-spans" tags:"state=dropped"`
+	ReporterDropped metrics.Counter `metric:"reporter_spans_dropped"`
 
 	// Current number of spans in the reporter queue
-	ReporterQueueLength metrics.Gauge `metric:"reporter-queue"`
+	ReporterQueueLength metrics.Gauge `metric:"reporter_queue_length"`
 
 	// Number of times the Sampler succeeded to retrieve sampling strategy
-	SamplerRetrieved metrics.Counter `metric:"sampler" tags:"state=retrieved"`
+	SamplerRetrieved metrics.Counter `metric:"sampler_retrieved"`
 
 	// Number of times the Sampler succeeded to retrieve and update sampling strategy
-	SamplerUpdated metrics.Counter `metric:"sampler" tags:"state=updated"`
+	SamplerUpdated metrics.Counter `metric:"sampler_updated"`
```

Comment:
"sampler_updates, result=ok" and "sampler_updates, result=err" to keep in line with the baggage stuff and we can update sampler_query_failures as you have.

Reply:
Still leaves sampler_retrieved and baggage_truncated ending in a verb.

Also, if we want to convert everything to end with a noun, we may consider s/spans_started/started_spans/ (finished_spans, etc.)